samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,241 @@
+    if (sprite == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create texture: %s\n", SDL_GetError());
+        SDL_FreeSurface(temp);
+        return NULL;
+    }
+    SDL_FreeSurface(temp);
+
+    /* We're ready to roll. :) */
+    return sprite;
+}
+
+void
+MoveSprites(SDL_Renderer * renderer, SDL_Texture * sprite)
+{
+    int sprite_w, sprite_h;
+    int i;
+    SDL_Rect viewport;
+    SDL_Rect *position, *velocity;
+
+    /* Query the sizes */
+    SDL_RenderGetViewport(renderer, &viewport);
+    SDL_QueryTexture(sprite, NULL, NULL, &sprite_w, &sprite_h);
+
+    /* Draw a gray background */
+    SDL_SetRenderDrawColor(renderer, 0x00, 0x00, 0x00, 0xFF);
+    SDL_RenderClear(renderer);
+
+    /* Move the sprite, bounce at the wall, and draw */
+    for (i = 0; i < NUM_SPRITES; ++i) {
+        position = &positions[i];
+        velocity = &velocities[i];
+        position->x += velocity->x;
+        if ((position->x < 0) || (position->x >= (viewport.w - sprite_w))) {
+            velocity->x = -velocity->x;
+            position->x += velocity->x;
+        }
+        position->y += velocity->y;
+        if ((position->y < 0) || (position->y >= (viewport.h - sprite_h))) {
+            velocity->y = -velocity->y;
+            position->y += velocity->y;
+        }
+
+        /* Blit the sprite onto the screen */
+        SDL_RenderCopy(renderer, sprite, NULL, position);
+    }
+
+    /* Update the screen! */
+    SDL_RenderPresent(renderer);
+}
+
+void demo(void)
+{
+    int i, done;
+    const char *driver;
+    SDL_Window *window;
+    SDL_Renderer *renderer;
+    SDL_Texture *sprite;
+    int window_w, window_h;
+    int sprite_w, sprite_h;
+    SDL_Event event;
+
+    /* Enable standard application logging */
+    SDL_LogSetPriority(SDL_LOG_CATEGORY_APPLICATION, SDL_LOG_PRIORITY_INFO);
+
+    if (SDL_VideoInit(NULL) < 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't initialize SDL video: %s\n",
+                SDL_GetError());
+        _exit(1);
+    }
+
+    if (TTF_Init() < 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't initialize TTF: %s\n",
+                TTF_GetError());
+        _exit(1);
+    }      
+    
+    driver = SDL_GetCurrentVideoDriver();
+
+    window = SDL_CreateWindow("Demo",
+        SDL_WINDOWPOS_UNDEFINED,
+        SDL_WINDOWPOS_UNDEFINED,
+        SCREEN_WIDTH, SCREEN_HEIGHT,
+        SDL_WINDOW_SHOWN);
+    if(window == NULL)
+    {
+        debugPrint( "Window could not be created! SDL_Error: %s\n", SDL_GetError() );
+        _exit(1);
+    }
+
+    /* Create the renderer */
+    renderer = SDL_CreateRenderer(window, -1, 0);
+    if (!renderer) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create renderer: %s\n", SDL_GetError());
+        _exit(5);
+    }
+
+    /* Clear the window, load the sprite and go! */
+    SDL_SetRenderDrawColor(renderer, 0xA0, 0xA0, 0xA0, 0xFF);
+    SDL_RenderClear(renderer);
```

Comment:
This block of code should probably be removed?
It's done in the renderer loop anyay; and when this is called, the program is still loading resources and doesn't even draw (I think). The code currently reads at `init init init *render init init loop: render`; so this `*render` block feels out of place.

So by the time it renders, this will have repeated anyway.